### ISSUE TITLE
Infinite recursion

### ISSUE THREAD

**Issue by benas** [benas/random-beans#44]
As of version 2.0.0-SNAPSHOT, when a type has a field of the same type, the populator enters an infinite loop which causes a `java.lang.StackOverflowError`. Here is an example:

```java
class Person {
   private String name;
   private Person parent;
}
```

**Comment by benas** [comment]
fixed in 592312b84fb8ec181a0e5d59720a7eb085739e6c

**Comment by benas** [comment]
related to #74
